front/pages/api/w/[wId]/mcp/request_access.ts

```diff
@@ -65,16 +64,31 @@ async function handler(
   }
 
   const emailRequester = user.email;
-  const { emailMessage, userTo } = bodyValidation.right;
+  const { emailMessage, mcpServerViewId } = bodyValidation.right;
 
-  const userRecipient = await UserResource.fetchById(userTo);
+  const mcpServerViewRes = await MCPServerViewResource.fetchById(
+    auth,
+    mcpServerViewId
+  );
 
-  if (!userRecipient) {
+  if (mcpServerViewRes.isErr()) {
     return apiError(req, res, {
       status_code: 404,
+      api_error: {
+        type: "mcp_server_view_not_found",
+        message: "The MCP server view was not found",
+      },
+    });
+  }
+
+  const mcpServerView = mcpServerViewRes.value;
+
+  if (!mcpServerView.editedByUser?.sId) {
+    return apiError(req, res, {
+      status_code: 403,
       api_error: {
         type: "user_not_found",
-        message: "The user was not found",
+        message: "No admin user found for this data source",
       },
     });
   }
```

Comment:
We're not trying to fetch the workspace admins[0] if there is no editedBy? Or the editedBy should already be setup by the first admin in 99% of the cases?

Reply:
There should always be one :+1: